pssh/ssh_client.py

```diff
@@ -335,3 +335,54 @@ def copy_file(self, local_file, remote_file, recurse=False):
         else:
             logger.info("Copied local file %s to remote destination %s:%s",
                         local_file, self.host, remote_file)
+
+    def _copy_dir_to_local(self, remote_dir, local_dir):
+        """Copies the remote directory to the local host."""
+        sftp = self._make_sftp()
+        file_list = sftp.listdir(remote_dir)
+        for file_name in file_list:
+            remote_path = os.path.join(remote_dir, file_name)
+            local_path = os.path.join(local_dir, file_name)
+            self.copy_file_to_local(remote_path, local_path, recurse=True)
+
+    def copy_file_to_local(self, remote_file, local_file, recurse=False):
+        """Copy remote file to local host via SFTP/SCP
+
+        Copy is done natively using SFTP/SCP version 2 protocol, no scp command \
+        is used or required.
+
+        :param remote_file: Remote filepath to copy the file from.
+        :type remote_file: str
+        :param local_file: Local filepath where the file will be copied.
+        :type local_file: str
+        :param recurse: Whether or not to recursively copy directories.
+        :type recurse: bool
+
+        :raises: :mod:'ValueError' when a directory is supplied to remote_file \
+        and recurse is not set
+        """
+        sftp = self._make_sftp()
+        try:
+            sftp.listdir(remote_file)
+            remote_dir_exists = True
+        except IOError or OSError:
+            remote_dir_exists = False
+        if remote_dir_exists and recurse:
+            return self._copy_dir_to_local(remote_file, local_file)
+        elif remote_dir_exists and not recurse:
+            raise ValueError("Recurse must be true if local_file is a "
+                             "directory.")
+        destination = [_dir for _dir in local_file.split(os.path.sep)
```

Comment:
My bad, you are correct - the intention is to check if directory paths in `local_file` exist.

Was forgetting why `destination` was added to `copy_file` in the first place :)

No, if local_file doesn't exist that doesn't necessarily mean the directory paths don't either so the existing logic is fine.

Might be worth moving the parent path split from filepath logic into its own function since it's being used in multiple places now (`copy_file` and now `copy_file_to_local`)